controllers/rating_controller.go

```diff
@@ -83,11 +83,16 @@ func (r *RatingReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return reconcile.Result{}, err
 	}
 
-	done, err := r.ratingChecker(ctx, &instance)
-	if !done {
+	update, err := r.ratingChecker(ctx, &instance)
+	if !update {
 		return reconcile.Result{Requeue: true}, err
 	}
 
+	done, err := r.enableratingChecker(ctx, &instance)
```

Comment:
@zqq454224016 you can keep this `enbale rating check` in ratingChecker. And do  `create pipelinerun` only when `rating enabled`

```golang
if instance.Spec.EnableRating {
    	if err := r.CreatePipelineRun(logger, ctx, &instance); err != nil {
		logger.Error(err, "")
		return reconcile.Result{}, err

}
}
```